cpp/src/arrow/compute/exec/asof_join_node_test.cc

```diff
@@ -83,6 +85,7 @@ Result<BatchesWithSchema> MakeBatchesFromNumString(
   batches.schema = schema;
   int n_fields = schema->num_fields();
   for (auto num_batch : num_batches.batches) {
+    Datum two(ConstantArrayGenerator::Int32(num_batch.length, 2));
```

Comment:
Why not a scalar here? (and remove `#include "arrow/testing/generator.h"` above)
```suggestion
    Datum two(Int32Scalar(2));
```